mlflow/tracking/fluent.py

```diff
@@ -582,6 +590,18 @@ def log_param(key: str, value: Any) -> Any:
     :param value: Parameter value (string, but will be string-ified if not).
                   All backend stores support values up to length 500, but some
                   may support larger values.
+    :param synchronous: Indicates if the metric would be logged in synchronous fashion or not.
+                    When it is true this call would be blocking call and offers immediate
+                        consistency of the metric upon returning.
+                    When this value is set to false, metric would be logged in async fashion.
+                    So backing provider gurantees that metrics are accepted into system
+                        but would persist them with some time delay.
+                    This means metric value would not have immediate consistency but
+                    'near-real'/'eventual' consistency.
+                    Note that this is an experimental flag.
+                    Default implementation of this flag would offer calling the sync API using a
+                    background thread.
+                    Each provider who may choose implement this flag can override the default behavior.
 
     :return: the parameter value that is logged.
```

Comment:
What value is returned when `synchronous` is false? What about `true`? Can we add docs for each case to the various log_param, log_metric, etc. methods?

Reply:
My assumption is return value should be same although default implementation in abstract_store level does not return anything just like log_param does not return anything there. Let me know your thoughts.

Reply:
I updated the docs for all log_param(s)/log_metric(s)/set_tag(s) in here and in MlflowClient as well.